### DISCUSSION TITLE
PROGMEM in custom_confg.h?

### DISCUSSION THREAD

**Discussion by idavydov** [manna-harbour/miryoku#137] [Q&A] [answered]
I'm trying to enable some unicode characters (for kyria rev2) using the following `custom_config.h`:

```
enum unicode_names {
    O_UMLAUT_SM,
    O_UMLAUT_CA
};

const uint32_t PROGMEM unicode_map[] = {
    [O_UMLAUT_SM]  = 0x00F6,
    [O_UMLAUT_CA]  = 0x00D6,
};
```

And I get the following error during compilation:

```
Compiling: users/manna-harbour_miryoku/manna-harbour_miryoku.c                                     In file included from ./users/manna-harbour_miryoku/config.h:8:0,
                 from <command-line>:0:
./users/manna-harbour_miryoku/custom_config.h:14:24: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘unicode_map’
 const uint32_t PROGMEM unicode_map[] = {
                        ^
 [ERRORS]
 | 
 | 
 | 
gmake[1]: *** [builddefs/common_rules.mk:456: .build/obj_splitkb_kyria_rev2_manna-harbour_miryoku/manna-harbour_miryoku.o] Error 1
Make finished with errors
gmake: *** [Makefile:413: splitkb/kyria/rev2:manna-harbour_miryoku:flash] Error 1
```

Any idea why `PROGMEM` could be undefined? (Not sure if that's the problem)

**Comment by manna-harbour** [comment, marked as answer]
Actual code, rather than just macro definitions, should go in a `keymap.c` or equivalent. `custom_keymap.c` will be added for this purpose for global use, see https://github.com/manna-harbour/miryoku/issues/94. At the moment you can use `manna-harbour_miryoku.c`, or if it's only for kyria, `keyboards/splitkb/kyria/keymaps/manna-harbour_miryoku/keymap.c`.

If you still have errors, please share a link to your branch.
